search/processor.cpp

```diff
@@ -608,15 +609,28 @@ bool Processor::SearchCoordinates()
 {
   bool coords_found = false;
   buffer_vector<ms::LatLon, 3> results;
+  double lat, lon;
 
+  if (MatchLatLonDegree(m_query, lat, lon))
   {
-    double lat;
-    double lon;
-    if (MatchLatLonDegree(m_query, lat, lon))
-    {
-      coords_found = true;
-      results.emplace_back(lat, lon);
-    }
+    coords_found = true;
+    results.emplace_back(lat, lon);
+  }
+
+  optional<ms::LatLon> maybeLatLon = MatchUTMCoords(m_query);
+  if (maybeLatLon)
+  {
+    coords_found = true;
+    auto latLon = maybeLatLon.value();
+    results.emplace_back(latLon.m_lat, latLon.m_lon);
```

Comment:
```
auto ll = ...;
if (ll)
  emplace_back(ll->m_lat, ll->m_lon);
```

Reply:
Fixed. Using `->` operator could be little slower 'cause it unpacks optional. But this code is not used often.